File: app/src/main/java/com/tk4218/grocerylistr/CustomLayout/CalendarDayAdapter.java
package com.tk4218.grocerylistr.CustomLayout;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.AsyncTask;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.tk4218.grocerylistr.Database.QueryBuilder;
import com.tk4218.grocerylistr.Model.ApplicationSettings;
import com.tk4218.grocerylistr.Model.Recipe;
import com.tk4218.grocerylistr.R;
import com.tk4218.grocerylistr.RecipeActivity;

import java.util.ArrayList;
import java.util.Date;

public class CalendarDayAdapter extends BaseAdapter {

    private ApplicationSettings mSettings;
    private Context mContext;
    private ArrayList<Recipe> mRecipes;
    private Date mCalendarDate;

    CalendarDayAdapter(Context context, ArrayList<Recipe> recipes, Date calendarDate){
        mSettings = new ApplicationSettings(context);
        mContext = context;
        mRecipes = recipes;
        mCalendarDate = calendarDate;
    }

    @Override
    public int getCount() {
        return mRecipes.size();
    }

    @Override
    public Recipe getItem(int position) {
        return mRecipes.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            final LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.listview_day_meals, parent, false);
        }

        TextView recipeName = convertView.findViewById(R.id.list_day_meal);
        recipeName.setText(mRecipes.get(position).getRecipeName());
        recipeName.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, RecipeActivity.class);
                intent.putExtra("recipeKey", mRecipes.get(position).getRecipeKey());
                mContext.startActivity(intent);
            }
        });

        recipeName.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                deleteMealPlan(mRecipes.get(position), position);
                return true;
            }
        });

        return convertView;
    }

    private void deleteMealPlan(final Recipe recipe, final int position){
        AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
        builder.setTitle("Un-schedule Recipe from Calendar")
                .setMessage("Are you sure you want to remove " + recipe.getRecipeName() + " from your calendar?")
                .setPositiveButton("Remove", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        new DeleteMealPlan().execute(mCalendarDate, recipe.getRecipeKey());
                        mRecipes.remove(position);
                        notifyDataSetChanged();
                    }
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    private class DeleteMealPlan extends AsyncTask<Object, Void, Void> {
        private QueryBuilder mQb = new QueryBuilder();
        @Override
        protected Void doInBackground(Object... params) {
            Date mealPlanDate = (Date) params[0];
            int recipeKey = (int) params[1];

            mQb.deleteCalendarRecipe(mSettings.getUser(), mealPlanDate, recipeKey);
            return null;
        }
    }
}

File: app/src/main/java/com/tk4218/grocerylistr/customlayout/CalendarDayAdapter.kt
package com.tk4218.grocerylistr.customlayout

import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.tk4218.grocerylistr.model.Recipe
import com.tk4218.grocerylistr.R
import com.tk4218.grocerylistr.RecipeActivity
import com.tk4218.grocerylistr.model.CalendarRecipes
import com.tk4218.grocerylistr.model.User

import java.util.ArrayList
import java.util.Date

class CalendarDayAdapter internal constructor(private val mContext: Context?, private val mRecipes: ArrayList<Recipe>, private val mCalendarDate: Date) : BaseAdapter() {
    override fun getCount(): Int {
        return mRecipes.size
    }

    override fun getItem(position: Int): Recipe {
        return mRecipes[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            val inflater = mContext?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.listview_day_meals, parent, false)
        }

        val recipeName = view!!.findViewById<TextView>(R.id.list_day_meal)
        recipeName.text = mRecipes[position].recipeName
        recipeName.setOnClickListener {
            val intent = Intent(mContext, RecipeActivity::class.java)
            intent.putExtra("recipeKey", mRecipes[position].recipeKey)
            mContext?.startActivity(intent)
        }

        recipeName.setOnLongClickListener {
            deleteMealPlan(mRecipes[position], position)
            true
        }

        return view
    }

    private fun deleteMealPlan(recipe: Recipe, position: Int) {
        val builder = AlertDialog.Builder(mContext)
        builder.setTitle("Un-schedule Recipe from Calendar")
                .setMessage("Are you sure you want to remove " + recipe.recipeName + " from your calendar?")
                .setPositiveButton("Remove") { _, _ ->
                    CalendarRecipes.removeRecipe(User.currentUsername(mContext!!), mCalendarDate, recipe.recipeKey)
                    mRecipes.removeAt(position)
                    notifyDataSetChanged()
                }
                .setNegativeButton("Cancel", null)
                .show()
    }
}
